packages/module/src/hooks/selection/useSelectionState.ts

```diff
@@ -1,11 +1,32 @@
 import React from 'react';
-import { DiscriminatedArgs } from '../../type-utils';
-import { ItemId } from '../../types';
+import { FeatureStateCommonArgs, ItemId } from '../../types';
+
+/**
+ * Feature-specific args for useSelectionState
+ * - Used as the `selection` sub-object in args of both useSelectionState and useTableState as a whole
+ * - Also included in the `TableBatteries` object returned by useTablePropHelpers and useClientTableBatteries.
+ * - Note that selection does not support the persistTo arg because it would break the derived state logic
+ *   (and because persisted checkbox selections are generally not desirable).
+ * @see UseTableStateArgs
+ * @see TableBatteries
+ */
+export interface SelectionStateArgs<TItem> extends Omit<FeatureStateCommonArgs, 'persistTo'> {
+  /**
+   * Ids of items to have pre-selected on first render
+   */
+  initialSelectedItemIds?: ItemId[];
+  /**
+   * Callback to determine if a given item is allowed to be selected. Blocks that item from being present in state.
+   * Added here even though it's not used in useSelectionState so args can all be passed at once. Actually used only in useSelectionDerivedState.
+   */
+  isItemSelectable?: (item: TItem) => boolean;
```

Comment:
Well, the `() => true` initializations are defaults/fallbacks for when `isItemSelectable` isn't passed at all, so code can assume it exists and it will just let all items be selectable. There's no reason for the consumer to pass a `isItemSelectable: true` at the feature level, which would just mean all items are selectable (which is the default behavior).